driver-core/src/main/com/mongodb/client/model/expressions/Expressions.java

```diff
@@ -81,4 +100,27 @@ public static ArrayExpression<BooleanExpression> ofBooleanArray(final boolean...
         return new MqlExpression<>((cr) -> new BsonArray(result));
     }
 
+
+    public static ArrayExpression<IntegerExpression> ofIntegerArray(final int... ofIntegerArray) {
+        List<BsonValue> array = Arrays.stream(ofIntegerArray)
+                .mapToObj(BsonInt32::new)
+                .collect(Collectors.toList());
+        return new MqlExpression<>((cr) -> new BsonArray(array));
+    }
+
+    public static DocumentExpression ofDocument(final Bson document) {
+        if (document == null) {
+            throw new IllegalArgumentException("document cannot be null");
+        }
+        // All documents are wrapped in a $literal. It is possible to check
+        // for empty docs (see https://jira.mongodb.org/browse/SERVER-46422)
+        // and to traverse for $$, but doing so would be unsafe and brittle.
```

Comment:
Could you please help me to understand this comment? Given the https://jira.mongodb.org/browse/SERVER-46422 link, I am guessing that all values are wrapped in `$literal` because this was one may `$addFields` with an empty document. Is this correct?

Regardless of whether the above guess is correct, I also need help with the following questions:

- What does _"to check for empty docs and to traverse for `$$`"_ mean and how is it related to creating a document expression?
- What does _"unsafe"_ mean in this context?

Reply:
Updated the comment. I don't want to dynamically detect whether a document needs to be wrapped in a literal. Instead, we should wrap all docs.